diana-api/src/main/java/org/jnosql/diana/api/reader/LocalDateReader.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.jnosql.diana.api.reader;
+
+
+import org.jnosql.diana.api.ReaderField;
+
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.Date;
+
+/**
+ * Class to reads and converts to {@link LocalDate} type
+ *
+ * @author Daniel Cunha - danielsoro@apache.org
+ */
+public final class LocalDateReader implements ReaderField {
+
+    @Override
+    public boolean isCompatible(Class clazz) {
+        return LocalDate.class.equals(clazz);
+    }
+
+    @Override
+    public <T> T read(Class<T> clazz, Object value) {
+
+        if (LocalDate.class.isInstance(value)) {
+            return (T) value;
+        }
+
+        if (Number.class.isInstance(value)) {
+            return (T) new Date(((Number) value).longValue()).toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
+        }
+
+        return (T) new Date(value.toString()).toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
```

Comment:
@danielsoro WDYT if we use `LocalDate.parse` instead?:
https://docs.oracle.com/javase/8/docs/api/java/time/LocalDateTime.html#parse-java.lang.CharSequence-